### DISCUSSION TITLE
SECRET_KEY and upgrade v3.4.x to v3.5.0

### DISCUSSION THREAD

**Discussion by cs-1** [netbox-community/netbox#12374] [Q&A] [answered]
I'd like to upgrade to v3.5.0 but looking at the breaking changes list, using a 50 char minimum length is now enforced. We now use 45 chars. Checking Django docs on the web, changing the key only breaks open sessions so I assume that simply changing the key to a longer one won't do any harm to NetBox. Can anyone confirm that this is the case and nothing will break in other places in NetBox when changing the SECRET_KEY parameter and restarting netbox / netbox-sq? Thanks for your help!

**Comment by abhi1693** [comment, marked as answer]
Nothing will break.

**Comment by cs-1** [comment]
Yes, I tested on my staging system that indeed everything will work. For all people who migrate: creating a new key with

```
python3 /opt/netbox/netbox/generate_secret_key.py
```

will give you a key with the proper length.

**Comment by ekrichbaum** [comment]
Yes. I just changed mine last night with no ill effects.

 

From: cs-1 ***@***.***> 
Sent: Friday, April 28, 2023 12:46 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] SECRET_KEY and upgrade v3.4.x to v3.5.0 (Discussion #12374)

 

I'd like to upgrade to v3.5.0 but looking at the breaking changes list, using a 50 char minimum length is now enforced. We now use 45 chars. Checking Django docs on the web, changing the key only breaks open sessions so I assume that simply changing the key to a longer one won't do any harm to NetBox. Can anyone confirm that this is the case and nothing will break in other places in NetBox when changing the SECRET_KEY parameter and restarting netbox / netbox-sq? Thanks for your help!

—
Reply to this email directly, view it on GitHub <https://github.com/netbox-community/netbox/discussions/12374> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACY47TLOJFQ5RWPCKVMZC3DXDNKQJANCNFSM6AAAAAAXOYOSVY> .
You are receiving this because you are subscribed to this thread.  <https://github.com/notifications/beacon/ACY47TMPAP4J7UAHKOBLKL3XDNKQJA5CNFSM6AAAAAAXOYOSV2WGG33NNVSW45C7OR4XAZNKIRUXGY3VONZWS33OVJRW63LNMVXHIX3JMTHAATTL7A.gif> Message ID: ***@***.*** ***@***.***> >



-- 
This email has been checked for viruses by Avast antivirus software.
www.avast.com
